pkg/glanceapi/statefulset.go

```diff
@@ -290,19 +293,33 @@ func StatefulSet(
 		extraVolPropagation),
 		apiVolumes...)
 
-	// If possible two pods of the same service should not
-	// run on the same worker node. If this is not possible
-	// the get still created on the same worker node.
-	statefulset.Spec.Template.Spec.Affinity = affinity.DistributePods(
-		common.AppSelector,
-		[]string{
-			glance.ServiceName,
-		},
-		corev1.LabelHostname,
-	)
 	if instance.Spec.NodeSelector != nil {
 		statefulset.Spec.Template.Spec.NodeSelector = *instance.Spec.NodeSelector
 	}
 
+	if topology != nil {
+		// Get the Topology .Spec
+		ts := topology.Spec
+		// Process TopologySpreadConstraints if defined in the referenced Topology
+		if ts.TopologySpreadConstraints != nil {
+			statefulset.Spec.Template.Spec.TopologySpreadConstraints = *topology.Spec.TopologySpreadConstraints
+		}
+		// Process Affinity if defined in the referenced Topology
+		if ts.Affinity != nil {
+			statefulset.Spec.Template.Spec.Affinity = ts.Affinity
+		}
+	} else {
+		// If possible two pods of the same service should not
+		// run on the same worker node. If this is not possible
+		// the get still created on the same worker node.
```

Comment:
do we have to explicite set the `statefulset.Spec.Template.Spec.TopologySpreadConstraints` to be empty in the case someone removed the ref back to default, since we update it using `CreateOrPatch`?

Reply:
No, if you remove the reference, a new `StatefulSet` is rolled out (because the `.Spec` changed), so we don't need to explicitly set that parameter to an empty `TopologySpreadConstraints` because it is already `nil` and `omitted` from the resulting `StatefulSet`.
The behavior is exactly the same we see with `NodeSelector`.

Reply:
cool, sounds good!